src/DependencyInjection/Configuration.php

```diff
@@ -52,7 +52,16 @@ private function addConverterSection(ArrayNodeDefinition $rootNode): void
                                 ->info('Mapping of source properties (value) to target properties (key)')
                                 ->normalizeKeys(false)
                                 ->useAttributeAsKey('target')
-                                ->prototype('scalar')->end()
+                                ->arrayPrototype()
+                                    ->beforeNormalization()
+                                    ->ifString()
+                                        ->then(fn (string $v) => ['source' => $v, 'default' => null])
+                                    ->end()
```

Comment:
Does this also work with `null` value?
Currently, we allow setting `~` as value when source and target are the same.
I assume we need to add another `ifNull()->then(` setting both to `null`?

Reply:
It works with `null` values. In this scenario, the Configuration returns an array where the key is the `target` propety and its corresponding value is an empty array. When the Converter attempts to access the `source` key, it encounters a failure and resorts to using the target property. Similarly, if the default value is set, it experiences the same failure and defaults to null. However, employing `ifNull` explicitly would be a superior approach. It results in code that is both easier to comprehend and more resilient, eliminating the need to rely on `??` to handle error cases."